module/scripts/combat/applyDamage.js

```diff
@@ -156,7 +156,7 @@ async function applyDamage(actor, dialogData, totalDamage, damageObject, derived
         totalDamage -= shield;
     }
 
-    if (damageObject.damageProperties.oilEffect === actor.system.category) {
+    if (damageObject.damageProperties.oilEffect && actor.system.category && (damageObject.damageProperties.oilEffect === actor.system.category)) {
```

Comment:
With the check of category the check for the oilEffect can be removed.

If a category is present and not null the oil effect needs to match it which it wont when it is null

Reply:
Fixed